shell/common/pipeline.h

```diff
@@ -24,6 +24,7 @@ enum class PipelineConsumeResult {
 
 size_t GetNextPipelineTraceID();
 
+/// A thread-safe queue of resources.
```

Comment:
"A single producer, single consumer resource queue which allows for book-keeping of frame workloads across multiple threads."

Reply:
Done.